test/unmanaged.cpp

```diff
@@ -0,0 +1,191 @@
+      exit(1);
+    }
+
+    DDS::PublisherQos pub_qos;
+    participant->get_default_publisher_qos(pub_qos);
+
+    DDS::Publisher_var pub =
+      participant->create_publisher(pub_qos, DDS::PublisherListener::_nil(),
+                                    ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (!pub) {
+      std::cerr << "create_publisher failed." << std::endl;
+      exit(1);
+    }
+
+    DDS::DataWriterQos datawriter_qos;
+    pub->get_default_datawriter_qos(datawriter_qos);
+
+    DDS::DataWriter_var basic_dw =
+      pub->create_datawriter(basic_topic,
+                             datawriter_qos,
+                             DDS::DataWriterListener::_nil(),
+                             ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!basic_dw) {
+      std::cerr << "create_datawriter failed for '" << basic_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    test::BasicMessageDataWriter_var basic_message_dw =
+      test::BasicMessageDataWriter::_narrow(basic_dw);
+
+    if (!basic_message_dw) {
+      std::cerr << "test::BasicMessageDataWriter::_narrow() failed." << std::endl;
+      exit(1);
+    }
+
+    DDS::DataWriter_var complex_dw =
+      pub->create_datawriter(complex_topic,
+                             datawriter_qos,
+                             DDS::DataWriterListener::_nil(),
+                             ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!complex_dw) {
+      std::cerr << "create_datawriter failed for '" << complex_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    test::ComplexMessageDataWriter_var complex_message_dw =
+      test::ComplexMessageDataWriter::_narrow(complex_dw);
+
+    if (!complex_message_dw) {
+      std::cerr << "test::ComplexMessageDataWriter::_narrow() failed." << std::endl;
+      exit(1);
+    }
+
+    bool run = true;
+
+    const std::string id = std::string(argv[0]) + '-' + to_str(std::this_thread::get_id());
+    CORBA::Long count = 0;
+
+    std::thread thread([&](){
+      while (run) {
+        test::BasicMessage basic_message{};
+        basic_message.origin = id.c_str();
+        basic_message.count = count;
+        basic_message_dw->write(basic_message, DDS::HANDLE_NIL);
+
+        test::ComplexMessage complex_message{};
+        complex_message.origin = id.c_str();
+        complex_message.count = count;
+        complex_message.ct.cuts.length(1);
+        test::TreeNode tn;
+        tn.ut.str("a string");
+        tn.et = test::EnumType::three;
+        complex_message.ct.cuts[0].tn(tn);
+        complex_message_dw->write(complex_message, DDS::HANDLE_NIL);
+
+        ++count;
+
+        std::this_thread::sleep_for(std::chrono::seconds(5));
+      }
+    });
+
+    std::string line;
+    std::getline(std::cin, line);
+
+    run = false;
+
+    thread.join();
+
+    participant->delete_contained_entities();
+    dpf->delete_participant(participant);
+  }
+  catch (const CORBA::Exception& e)
+  {
```

Comment:
```suggestion
  } catch (const CORBA::Exception& e)  {
```

Reply:
Since it's in a test and in another repo I wasn't attempting to enforce style guidelines